cmd/flush_handles_test.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/google/go-tpm-tools/simulator"
+	"github.com/google/go-tpm-tools/tpm2tools"
+	"github.com/google/go-tpm-tools/tpm2tools/tpm2toolstest"
+	"github.com/google/go-tpm/tpm2"
+)
+
+const (
+	// Maximum number of handles to keys tests can create within a simulator.
+	maxHandles = 3
+)
+
+func TestHandleTypesFromFlags(t *testing.T) {
+	for _, test := range []struct {
+		flushTransientFlag     bool
+		flushLoadedSessionFlag bool
+		flushSavedSessionFlag  bool
+		flushAllTypesFlag      bool
+		want                   []tpm2.HandleType
+	}{
+		{false, false, false, false, []tpm2.HandleType{}},
+		{true, false, false, false, []tpm2.HandleType{tpm2.HandleTypeTransient}},
+		{false, true, false, false, []tpm2.HandleType{tpm2.HandleTypeLoadedSession}},
+		{false, false, true, false, []tpm2.HandleType{tpm2.HandleTypeSavedSession}},
+		{false, true, true, false, []tpm2.HandleType{tpm2.HandleTypeLoadedSession, tpm2.HandleTypeSavedSession}},
+		{true, true, false, false, []tpm2.HandleType{tpm2.HandleTypeTransient, tpm2.HandleTypeLoadedSession}},
+		{true, false, true, false, []tpm2.HandleType{tpm2.HandleTypeTransient, tpm2.HandleTypeSavedSession}},
+		{true, true, true, false, []tpm2.HandleType{tpm2.HandleTypeTransient, tpm2.HandleTypeLoadedSession, tpm2.HandleTypeSavedSession}},
+		{false, false, false, true, []tpm2.HandleType{tpm2.HandleTypeTransient, tpm2.HandleTypeLoadedSession, tpm2.HandleTypeSavedSession}},
+	} {
+		// Overwrite flag pointer values.
+		flushTransient = &test.flushTransientFlag
```

Comment:
The documentation on the flag package isn't clear about usage in test contexts. It implies the [flag.Parse() method](https://golang.org/pkg/flag/#Parse) must be used before accessing flags. I didn't want to call Parse() because I am not sure what the argvs would be at runtime during tests. 

I like your solution of using flag.Set since it is more readable. It just isn't very clear in the documentation that you can do this.

Reply:
TL;DR: avoid global state (including flags) whenever possible.

Generally speaking, tests shouldn't know about flags at all (unless you test flag parsing).
Flags are `func main`-level concern and all other code should get configuration passed explicitly through func arguments or struct fields.
Looking outside of their context into the global vars (which get set through flags at runtime) makes testing harder: you can't parallelize tests that read flags due to memory safety, you must remember to clean up modified values to make tests order-independent.

However, this program is not production-critical so we don't need to be so strict about code structure or testing.